content/learning-paths/mobile-graphics-and-gaming/android_halide/intro.md

```diff
@@ -0,0 +1,165 @@
+    Mat output(input.rows, input.cols, CV_8UC3, outputBuffer.data());
+
+    // Convert RGB back to BGR for proper display in OpenCV.
+    cvtColor(output, output, COLOR_RGB2BGR);
+
+    // Display the input and processed image.
+    imshow("Original Image", input);
+    imshow("Inverted Image", output);
+    waitKey(0); // Wait for a key press before closing the window.
+
+    return 0;
+}
+```
+
+This program demonstrates how to combine Halide’s image processing capabilities with OpenCV’s image I/O and display functionality. It begins by loading an image from disk using OpenCV, specifically reading from a static file named img.png (here we use a Cameraman image). Since OpenCV loads images in BGR format by default, the code immediately converts the image to RGB format so that it is compatible with Halide’s expectations.
+
+Once the image is loaded and converted, the program wraps the raw image data into a Halide buffer, which captures the image’s dimensions (width, height, and the number of color channels). An ImageParam is then created, allowing the Halide pipeline to use symbolic indexing with variables. The core of the pipeline is defined by a function called invert, which computes a new pixel value by subtracting the original value from 255. This operation effectively inverts the image’s colors. The scheduling directive invert.reorder(c, x, y) ensures that the output buffer is stored in an interleaved manner—making it compatible with the memory layout expected by OpenCV.
+
+After the pipeline processes the image, the output is realized into another Halide buffer. Instead of copying pixel data back and forth, the program directly wraps this buffer into an OpenCV Mat header. This efficient step avoids unnecessary data duplication. Finally, because OpenCV displays images in BGR format, the code converts the processed image back from RGB to BGR. The program then displays both the original and the inverted images in separate windows, waiting for a key press before exiting. This workflow showcases a streamlined integration between Halide for high-performance image processing and OpenCV for handling input and output operations.
+
+## Compilation Instructions
+Compile the program as follows (replace /path/to/halide accordingly):
+```console
+export DYLD_LIBRARY_PATH=/path/to/halide/lib/libHalide.19.dylib
+g++ -std=c++17 hello-world.cpp -o hello-world \
+    -I/path/to/halide/include -L/path/to/halide/lib -lHalide \
+    $(pkg-config --cflags --libs opencv4) -lpthread -ldl \
+    -Wl,-rpath,/path/to/halide/lib
+```
+
+Run the executable:
+```console
+./hello-world
+```
+
+You will see two windows displaying the original and inverted images:
+![img1](Figures/01.png)
+![img2](Figures/02.png)
+
+## Summary
+In this lesson, you’ve learned Halide’s foundational concepts, explored the benefits of separating algorithms and schedules, set up your development environment, and created your first functional Halide application integrated with OpenCV. 
```

Comment:
At this moment, "benefit of separating algorithms and schedules" is not shown very much with the example above. I suppose it would be shown in the later example.